File: movie-front-end/src/app/models/movie.ts
export type Movie={
    movieId: string;
    movieTitle: string;
    releaseYear:string;
    genre:string;
    movieImagePath:string;
    movieDescription:string;
    movieRating:number;
    movieDirector:string;
    movieWriters:string[];
    trailerLink:string;
    streamingOn:string[];
    castAndCrew:string[];
    availableLanguages:string[];
    movieReviews?:string[];
}

File: moviebackend/MovieService/src/main/java/com/example/MovieService/model/Movie.java
package com.example.MovieService.model;

import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;
import org.springframework.data.annotation.Id;
import org.springframework.data.mongodb.core.mapping.Document;

import java.util.List;

@AllArgsConstructor
@NoArgsConstructor
@Data
@Document
public class Movie {
    @Id
    private String movieId;
    private String movieTitle;
    private String releaseYear;
    private String genre;
    private String movieImagePath;
    private String movieDescription;
    private double movieRating;
    private List<String> movieReviews;
    private String movieDirector;
    private List<String> movieWriters;
    private String trailerLink;
    private List<String> streamingOn;
    private List<String> castAndCrew;
    private List<String> availableLanguages;


}
